### ISSUE TITLE
False-positives alerts for interface type conversions

### ISSUE THREAD

**Issue by Ragnar-BY** [kisielk/errcheck#259]
Last version golangci-lint use errcheck v1.8.0
Now errcheck gives alerts for code like:

```go
	var a interface{}
	a = 5

	switch a.(type) {
	case string:
		fmt.Println("this is string", a.(string))
	case int:
		fmt.Println("this is int", a.(int))
	default:
		fmt.Println("unknown type")
	}
```
I think it is related to changes from #246 
I believe that it is false-positive, because type is checked in switch-case

**Comment by dtcaciuc** [comment]
Current errcheck behavior is consistent with `gopls`. Recommended action is to reassign a to make it assume the resolved type
![gopls-type-assert](https://github.com/user-attachments/assets/38768684-03e5-46f1-991d-3fed0afec9dd)
